contracts/YieldStrategyLever.sol

```diff
@@ -0,0 +1,155 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.14;
+
+import "./YieldLeverBase.sol";
+
+interface IStrategy {
+    function mint(address to) external returns (uint256 minted);
+
+    function burn(address to) external returns (uint256 withdrawal);
+
+    function burnForBase(address to) external returns (uint256 withdrawal);
+
+    function pool() external returns (IPool pool);
+}
+
+contract YieldStrategyLever is YieldLeverBase {
+    using TransferHelper for IERC20;
+    using TransferHelper for IFYToken;
+
+    mapping(bytes6 => IStrategy) strategies;
+    address constant USDC = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
+    address constant DAI = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+
+    constructor(Giver giver_) YieldLeverBase(giver_) {
+        IERC20(USDC).approve(
+            address(LADLE.joins(0x303200000000)),
+            type(uint256).max
+        );
+        IERC20(DAI).approve(
+            address(LADLE.joins(0x303100000000)),
+            type(uint256).max
+        );
+    }
+
+    function setStrategy(bytes6 ilkId, IStrategy strategy) external {
```

Comment:
This function is unsafe, anyone could add or modify any strategy associated to any ilkId.

Why do we need this, more exactly? Please add the natspec as you reply.